textdb/textdb-api/src/test/java/edu/uci/ics/textdb/api/common/SchemaTest.java

```diff
@@ -43,4 +42,22 @@ public void testGetIndex(){
         Assert.assertEquals(expectedIndex_2, retrievedIndex_2);
         
     }
+    
+    @Test(expected = UnsupportedOperationException.class)
+    public void testAddingNewAttribute(){ //Should fail

```

Comment:
@chenlica Yes. 
But the immutability is not complete. We can still override an element present at particular index using `List.set()` method. This list is just like an `Array`.